File: Engine/Core/Resources/Texture/TextureManager.cpp
#include "TextureManager.h"

#include "TextureUtilities.h"

#include <glad/glad.h>

int TextureManager::loadTexture(const std::string& fileName, bool isOpaque){
    int format;
    if(isOpaque){
        format = GL_REPEAT;
    }else{
        format = GL_CLAMP_TO_EDGE;
    }
    return TextureUtils::createTexture(fileName.c_str(), GL_RGBA, format);
}

File: gamedata/ChunkLoader/Textures/TextureManager.lua
TextureManager = {}
TextureManager.textures = {}

TextureManager.loadTexture = function(this, fileName, textureName, isOpaque)
    local textureID = Engine.loadTexture(fileName, isOpaque)
    this.textures[textureName] = textureID
end
TextureManager.getTexture = function(this, textureName)
    return this.textures[textureName]
end
